### ISSUE TITLE
Create example demo app/

### ISSUE THREAD

**Issue by de-sh** [de-sh/paex#3]
The demo application making use of the template to execute a distributable task over the cluster and the various components necessary to manage disparate events across the cluster.

**Comment by de-sh** [comment]
@123vivekr, If I am not wrong we are building the demo app to apply a machine leaerning filters to the images and run processes on partitions of the image?

Can you further clarify what this app will require?

**Comment by de-sh** [comment]
What are your thoughts on doing this in an entirely different repo?

**Comment by bkp31415** [comment]
We could implement a simple case like, sending four pairs of numbers to four nodes and calculate their averages in response via grpc which we can then extend to images.

**Comment by de-sh** [comment]
Can you expand on this? I think you are hinting at a divide and conquer strategy for image compression?

**Comment by 123vivekr** [comment]
The main idea might sound simple but the implementation is quite complex. So I agree with @bkp31415 's method. Let's implement using the bottom up approach. :+1:

**Comment by 123vivekr** [comment]
@de-sh: @bkp31415 and I had a discussion about the implementation and we concluded with the idea of bottom up approach. Basically, we start with a simple implementation of the logic: Send 3 numbers to each of the node and get back the average of the 3.

So, if we have 4 nodes, then we send 3 numbers to each of them, so a total of 12. We then get back the result and compare with the hard-coded average, since we already know what those numbers are.

Then we scale up to more than 4 nodes and also scale down to less than 4. This is step 2.

It is all fine up to this, but what if a node fails..?
We borrow the idea of the Raft algorithm and ensure our system works in case of disruptions and failures.
We can test this by deliberately failing and/or disconnection a node.

**Comment by de-sh** [comment]
I am not sure if we have to manually or descriptively scale the processes, but I get your idea, each node can run multiple pods with each pod executing a basic calculation, then collating it into a result computation

**Comment by 123vivekr** [comment]
Yes, let's leave the scaling to kubernetes. I mentioned about it to have continuity. :)
